chromium_src/extensions/common/permissions/permissions_data.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "extensions/common/permissions/permissions_data.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+
+namespace extensions {
+
+bool IsBraveProtectedUrl(const url::Origin& origin, base::StringPiece path) {
```

Comment:
why not just pass the GURL here? It seems like we're making things difficult on the caller